pkg/services/ngalert/api/lotex_ruler.go

```diff
@@ -177,7 +217,13 @@ func (r *LotexRuler) RoutePostNameRulesConfig(ctx *contextmodel.ReqContext, conf
 	if err != nil {
 		return ErrResp(500, err, "Failed marshal rule group")
 	}
-	u := withPath(*ctx.Req.URL, fmt.Sprintf("%s/%s", legacyRulerPrefix, ns))
+
+	var finalNamespace = ns
+	if ns == "QUERY_NAMESPACE" {
+		finalNamespace = ctx.Query("namespace")
+	}
+
+	u := withPath(*ctx.Req.URL, fmt.Sprintf("%s/%s", legacyRulerPrefix, finalNamespace))
```

Comment:
```suggestion
	u := withPath(*ctx.Req.URL, fmt.Sprintf("%s/%s", legacyRulerPrefix, url.PathEscape(finalNamespace)))
```

Reply:
@konrad147 I think we still need to escape the input here?

Reply:
It's done